### PR TITLE
Adding tests for ReadOnlyProperty and Inheritance in complexgroup

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding tests for ReadOnlyProperty and Inheritance in complexgroup

### CODE REVIEW

File: test/autorest/complexgroup/readonlyproperty_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	expected := &complexgroup.ReadonlypropertyGetValidResponse{
+		StatusCode:  http.StatusOK,
+		ReadonlyObj: &complexgroup.ReadonlyObj{ID: &id, Size: &size},
+	}
+	deepEqualOrFatal(t, result, expected)
+}
+
+func TestReadonlypropertyPutValid(t *testing.T) {
+	client := getReadonlypropertyOperations(t)
+	result, err := client.PutValid(context.Background(), complexgroup.ReadonlyObj{})
```
Comment (jhendrixMSFT) [inline test/autorest/complexgroup/readonlyproperty_test.go]: So here the test should specify `ID` and `Size` (I assume these are both RO) and it should still pass.  Probably won't right now due to lack of custom marshalling to omit RO properties.  If you can confirm, probably want to open an issue and comment out the test.

Comment (catalinaperalta) [reply test/autorest/complexgroup/readonlyproperty_test.go]: Ah I see, thanks for pointing that out. I didn't understand what was happening with this test. Yes, we aren't doing the custom marshalling to omit the RO properties so they were being sent over to the endpoint which was rejecting them. I'll update the test and comment it out.

Comment (jhendrixMSFT) [reply test/autorest/complexgroup/readonlyproperty_test.go]: Might want to open an issue too.

Comment (catalinaperalta) [reply test/autorest/complexgroup/readonlyproperty_test.go]: Done. Here's the issue: #259

**Pull request merged**